api/v3/Grant.php

```diff
@@ -54,6 +54,7 @@ function civicrm_api3_grant_create($params) {
  *   Array of parameters determined by getfields.
  */
 function _civicrm_api3_grant_create_spec(&$params) {
+  $params['contact_id']['api.required'] = 1;
```

Comment:
@jitendrapurohit

According to the UI, yes they should.

The API call works even if they aren't defined. This is because they are both defaulted to 0 (though perhaps they shouldn't be).

It kind of makes sense in the case for the amount (since 0 is a valid amount) but not in the case of the status_id since 0 is not a valid status ID.

It appears that 0 will never get to _civicrm_api3_api_match_pseudoconstant as 0 evaluates to empty. Not sure if there is anything we can do about that but I'll raise a seperate issue, and add the status_id and amount_total as required for this API.